### DISCUSSION TITLE
CrewAI Templates

### DISCUSSION THREAD

**Discussion by theCyberTech** [crewAIInc/crewAI#1157] [Polls]
We're eager to hear your thoughts on what style of templates you would use for CrewAI.

**Comment by Jsin01** [comment]
Coding for the win!

**Reply by theCyberTech** [reply]
What sort of coding use cases would you have in mind?

**Reply by BarryBahrami** [reply]
It would be awesome to point it to online API or coding language docs, give it a task and let it have at it.  I understand we need to break these down but that's the gist of it. 

while it's at it, turn lead into gold.  It can do that, right?

**Comment by RobotMachinesAI** [comment]
I suggest creating a template of agents that act as a '**machine control crew**' to ensure machines follow command instructions accurately. This template would include agents for:

- command interpretation
- multiple task response capability
- tool use (vision)
- proper response syntax check (responses would be used for command control and the syntax will need to be exact)
- issue user/system feedback request upon uncertainty

It would benefit developers testing AI for automation and control systems.

**Comment by rmalmstead2021** [comment]
I would love to see a working code crew template, that adheres to an agile coding framework, like Scrum or Kanban.
The dream is... a crew working 24 hours a day on a backlog created by the product owner humans.  I've seen some repositories get close with langraph/langchain, and of course some paid online offerings, but so far not as impressive as it could be imo.  The backlog should be live in the RAG/memory, as should every page in the codebase itself.  I saw a repository that links directly into GitHub but I couldn't get it work myself... mostly because I don't want to pay for another subscription, I have enough already.

**Comment by jastingo** [comment]
I would love to see SQL coding templates related to the data craft - data modeler, data engineer, data analyst, etc.
